kext/FileNVRAM/FileNVRAM.cpp

```diff
@@ -115,49 +115,115 @@ bool FileNVRAM::start(IOService *provider)
     if(!dict) return false;
     setPropertyTable(dict);
     
-    
     if(bootnvram)
     {
         copyEntryProperties(NULL, bootnvram);
         bootnvram->detachFromParent(root, gIODTPlane);
+        // We restored the nvram from the bootloader. register the service
+        finalizeInit();
+        // waiting for IOBSD to determine if is safe to sync..
     }
     
     if(mReadOnly)
     {
         // we assume that the bootloader has done its job
         // i.e. populate /chosen/nvram
-        mInitComplete = true;
         mSafeToSync = true;
-        registerNVRAM();
+        if(!mInitComplete)
+        {
+            finalizeInit();
+        }
         return true;
     }
     else
     {
-        IOTimerEventSource* mTimer = IOTimerEventSource::timerEventSource(this, timeoutOccurred);
-        
-        if(mTimer)
+        // waiting for a client to appear before watching for IOBSD
+        if(VERSION_MAJOR >10)
```

Comment:
I'm not sure if this is what you want. This is a compile time constant ad has no correlation to the actual kernel that you are running on.